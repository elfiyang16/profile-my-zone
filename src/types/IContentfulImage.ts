export interface IContentfulImage {
  id?: string
  title: string
  altText?: string
  file: {
    url: string
    fileName?: string
    contentType?: string
  }
}
