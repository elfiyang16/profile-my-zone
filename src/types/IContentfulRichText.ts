import { ContentfulRichTextGatsbyReference } from "gatsby-source-contentful/rich-text"

export interface IContentfulRichText<
  T extends ContentfulRichTextGatsbyReference
> {
  raw: string
  references: T[]
}
