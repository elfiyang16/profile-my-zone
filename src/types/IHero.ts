import {IContentfulImage} from "./IContentfulImage"

export interface IHero {
  colorTheme?: string
  heading1?: string
  heading2? :string
  image?:IContentfulImage
}
