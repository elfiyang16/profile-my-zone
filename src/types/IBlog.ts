import { IContentfulRichText } from "./IContentfulRichText"
import { IContentfulImage } from "./IContentfulImage"

export interface IBlog {
  id?: string
  author?: { name: string }
  slug?: string
  title: string
  description?: {
    description: string
  }
  body: IContentfulRichText
  publishDate?: string
  heroImage?: IContentfulImage
}
