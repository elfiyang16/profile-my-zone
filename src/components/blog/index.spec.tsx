import { shallow } from "enzyme"
import React from "react"
import Blog from "."
import { IBlog } from "../../types/IBlog"
export const mockBlog: IBlog = {
  title: "Python Concurrency (i)",
  publishDate: "2020-11-22T00:00+00:00",
  heroImage: {
    title: "Photo on Unsplash",
    file: {
      url: "//images.ctfassets.net/abe.jpeg",
    },
  },
  body: {
    raw: '{"nodeType":"document","data":{},"content":[]}',
  },
  description: {
    description: "description",
  },
  author: { name: "elfi" },
  slug: "/abv",
  id: "1234",
}

describe("<Blog />", () => {
  let wrapper: any
  beforeEach(() => {
    wrapper = shallow(<Blog {...mockBlog} />)
  })

  it("should render", () => {
    expect(wrapper).toMatchSnapshot()
  })
})
