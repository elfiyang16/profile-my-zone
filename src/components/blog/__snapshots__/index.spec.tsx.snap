// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog /> should render 1`] = `
<div
  className="article"
>
  <div
    className="article__img-container"
  >
    <img
      alt="Photo on Unsplash"
      className="article__img"
      src="//images.ctfassets.net/abe.jpeg"
    />
    <small
      className="article__img-title"
    >
      Photo on Unsplash
    </small>
  </div>
  <article
    className="article__content"
  >
    <h2
      className="article__content-title"
    >
      Python Concurrency (i)
    </h2>
    <p
      className="article__content-date"
    >
      November 22, 2020
    </p>
    <Component
      content={
        Object {
          "raw": "{\\"nodeType\\":\\"document\\",\\"data\\":{},\\"content\\":[]}",
        }
      }
    />
  </article>
</div>
`;
